onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -28,7 +28,7 @@ def testModelSerialization(self):
             so.log_verbosity_level = 1
             so.logid = "TestModelSerialization"
             so.optimized_model_filepath = "./PythonApiTestOptimizedModel.onnx"
-            onnxrt.InferenceSession(get_name("mul_1.onnx"), sess_options=so)
+            onnxrt.InferenceSession(get_name("mul_1.onnx"), sess_options=so, providers=['CPUExecutionProvider'])
```

Comment:
why do we need to register CPU provider?

Reply:
it's just a test of model serialization, so CPU provider should be sufficient, right?
it loads mul_1.onnx and writes PythonApiTestOptimizedModel.onnx and checks the file was written.
we can't leave providers empty, since we would raise exception for a gpu build.

Reply:
So, if someone is using the gpu pkg on a CPU-only machine, they'll have to register the CPU provider explicitly? Should the CPU not be the default such that it doesn't require an explicit registration?

Reply:
Talked to George. He'll document this behavior.